### DISCUSSION TITLE
Ideas: Generate generic static website in Drasil

### DISCUSSION THREAD

**Discussion by cd155** [JacquesCarette/Drasil#3010] [Ideas]
Back to summer 2021, we completely move from Hakyll to Drasil HTML Generator in #2638

Looking at the `drasil-website/lib/Drasil/Website` shows the capability of the Drasil HTML generator is capable of generating a generic static website. I think this promotes the idea, that Drasil can be used in another area rather than scientific research area.

One example I can think of is generating a static website for personal blog use. This will make Drasil comparable with Hakyll. Able to generate other code also is a great advantage Drasil has. Some educational coding websites usually will insert code into HTML pages for many reasons, like explaining how this code work.

The purpose of this ticket is just to brainstorm how Drasil can be used in other areas.

**Comment by smiths** [comment]
Drasil can definitely be used in areas other than research software.  The only real requirement for Drasil is that the domain is well understood.  The only reason we've emphasized research software is that it fits the criteria of being well-understood, we understand it :smile:, and the research community could greatly benefit.  (The research software community doesn't yet recognize the last point, but I believe it is true nonetheless.)  :smile:

@tingyuw is working on Drasil to generate Jupyter notebooks.  When she gets that working, we'll have the ability to mix documentation and code.  We could then use this to generate other documents that mix documentation and code, like org mode documents, or the webpages you describe.

Thank you @cd155 for posting this discussion topic.

**Reply by cd155** [reply]
No problem. I used to build my own website, and recently try to build the website with Haskell, then it come to the Hakyll library. After looking around at the Drasil website, I found a static website can be built in Drasil because Drasil can generate HTML. Somehow, after I finish the final report, I would like to use Drasil to generate a static website (e.g. a blog) and see how it works. Is this action allowed? I just want to make sure there is no licence issue for the future plan.

**Reply by cd155** [reply]
I completely forgot that one big advantage of libraries like Hakyll is that they allow writing articles in Markdown. It will increase the readability so much. Using the Drasil to generate HTML could lose the advantage of writing in Markdown because it is less readable in general compare to Markdown.

**Reply by JacquesCarette** [reply]
This is definitely allowed.

And making it be as nice as writing Markdown would be the purpose of having an external syntax for parts of Drasil, instead of embedding it all inside Haskell as we currently do.

I think that org-mode is a reasonable example of what that might look like.

**Comment by smiths** [comment]
@cd155, the [license agreement](https://github.com/JacquesCarette/Drasil/blob/master/LICENSE) for Drasil certainly allows for forks.  You can certainly experiment with using it for a static website.  Although not required by the license, please let us know about any successes you have.  :smile:

If your goal is to make writing a web-page as easy as markdown, Drasil probably isn't the right choice.  If the goal is to generate multiple artifacts as different views of the same knowledge, Drasil is a great choice.  Likewise, if you wish to be able to change high-level decisions (analogous to logging on/off), then Drasil is a good choice.  :smile:

**Reply by cd155** [reply]
Surprisingly, the license file was added back in 2014.

**Reply by smiths** [reply]
Why is that surprising?  Is it because the copyright notice isn't current?  I don't think the date refers to the software as of that date, but rather the date the license was written.

**Reply by cd155** [reply]
I was surprised that the Drasil project may start at least in 2014 because the license file commits in 2014. It is 2022 now, the time has been passing very quickly.

**Reply by JacquesCarette** [reply]
We have been working on this for a long time, that's true. Drasil of 2016, while it did generate a lot, would likely be unrecognizable to you.

**Comment by smiths** [comment]
Yes, 2014 is a long time ago!  Out of curiosity, I found the first commit:

https://github.com/JacquesCarette/Drasil/commit/e2491ae589ae48381983fec936b9d75baedd0706

The license was in the first commit.  :smile:
